### ISSUE TITLE
Fedora 30 template

### ISSUE THREAD

**Issue by fepitre** [QubesOS/qubes-issues#4845]
Tracking issue for fedora-30 template:
- [x] build all packages
- [x] build the template
- [x] document
- [x] upload to testing repo
- [x] migrate to stable repo
- [ ] announce

**Comment by marmarek** [comment]
@fepitre does all packages build for fc30? Is it the time to enable building packages for fc30 in release config?

**Comment by fepitre** [comment]
Two weeks ago it built fine. Only few dom0 packages still need work.

**Comment by fepitre** [comment]
I have not referenced all the commits on the builder-rpm but it's merged. I can retry a build tomorrow if you want.

**Comment by marmarek** [comment]
[fedora-30 template](https://github.com/QubesOS/updates-status/issues/1041) is in templates-itl-testing repository already.
Now is the time for some testing, and also updating documentation. @fepitre @andrewdavidwong will you take care of it?

I've tried [recommended DNF system-upgrade](https://fedoraproject.org/wiki/DNF_system_upgrade) method, but there are a couple problems with it on Qubes:
 - you need to disable `qrexec` feature (`qvm-features -D fedora-30 qrexec`) before the operation, otherwise the template will get killed during update, as it is performed as part of system startup
 - `qrexec-agent` isn't running during the operation, which means application list and other settings are not sent to dom0 automatically - one needs to call `/etc/qubes-rpc/qubes.PostInstall` in the template manually after the upgrade
 - reboot from inside VM really shut it down - user needs to start the template manually when reboot was requested
 - there is no visible progress, as it happens at early VM startup and we don't show VM console by default; one can observe `sudo xl console fedora-30`
 - directly after upgrade (when `qrexec` feature is still disabled), `qvm-run` may fail - one needs to wait until VM is fully started and only then call `qvm-run` (this applies to menu entries too); `/etc/qubes-rpc/qubes.PostInstall` will announce to dom0 `qrexec` support, so it won't affect further startups
 - (independent of the method) rpmfusion problem still applies

At this point, I'm not sure if the above problems worth describing them, or better stick with the old method we have documented already.
For the future, it may be good to automate at least some of those steps. For example add a mechanism VM can use to tell dom0 the update is running (or that it will be at the next VM start). Then, dom0 would not kill it in the process, and also may display some notification what is happening.

**Comment by andrewdavidwong** [comment]
> Now is the time for some testing, and also updating documentation. @fepitre @andrewdavidwong will you take care of it?

Sure thing; I'll work on the documentation.

> I've tried recommended DNF system-upgrade method, but there are a couple problems with it on Qubes:
> [...]
> At this point, I'm not sure if the above problems worth describing them, or better stick with the old method we have documented already.

Wow, that's a long list of problems. Sounds like we should just stick with the old method, if it's no worse.

**Comment by fepitre** [comment]
> For the future, it may be good to automate at least some of those steps. For example add a mechanism VM can use to tell dom0 the update is running (or that it will be at the next VM start). Then, dom0 would not kill it in the process, and also may display some notification what is happening.

I will test your steps and see what I can do easily with that.

**Comment by andrewdavidwong** [comment]
The upgrade documentation has been merged.

**Comment by andrewdavidwong** [comment]
I'm confused: https://ftp.qubes-os.org/repo/yum/r4.0/templates-itl/rpm/ shows that the Fedora 30 template has been available for a whole month, yet @fepitre was still testing as recently as 8 days ago, and the `migrate to stable repo` step in the description is still unchecked. Was it uploaded to the stable repo by mistake?

CC: @marmarek

**Comment by marmarek** [comment]
> I'm confused: https://ftp.qubes-os.org/repo/yum/r4.0/templates-itl/rpm/ shows that the Fedora 30 template has been available for a whole month

That's template build date, not migration to stable. You'll have better details here: https://github.com/QubesOS/updates-status/issues/1041. I've migrated it to stable after testing by @fepitre

**Comment by andrewdavidwong** [comment]
> > I'm confused: https://ftp.qubes-os.org/repo/yum/r4.0/templates-itl/rpm/ shows that the Fedora 30 template has been available for a whole month
> 
> That's template build date, not migration to stable. You'll have better details here: [QubesOS/updates-status#1041](https://github.com/QubesOS/updates-status/issues/1041). I've migrated it to stable after testing by @fepitre

Oh. Did you forget to check the `migrate to stable repo` box when you migrated it to stable? I saw that it was unchecked, so I reported that it was still in testing when someone asked, then Lorenzo pointed out that it was actually in stable:

https://groups.google.com/d/topic/qubes-users/h1LCKNBOaKw/discussion

I believe it is important to keep checklist items (and issues in general) updated so that people can reliably refer to them for the current status on things and to avoid confusion.

(Now that it's confirmed that we intend for it to be in stable, I'll check the box.)
